src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeEntryProcessor.java

```diff
@@ -0,0 +1,87 @@
+
+    private final DateTimeFormatParser dateTimeFormatParser;
+
+    @Autowired
+    public DateTimeEntryProcessor(CaseViewFieldBuilder caseViewFieldBuilder,
+                                  DateTimeFormatParser dateTimeFormatParser) {
+        super(caseViewFieldBuilder);
+        this.dateTimeFormatParser = dateTimeFormatParser;
+    }
+
+    @Override
+    protected JsonNode executeSimple(JsonNode node, CommonField field, BaseType baseType, String fieldPath) {
+        return !isNullOrEmpty(node)
+            && hasDisplayContextParameterType(field.getDisplayContextParameter(), DisplayContextParameterType.DATETIMEENTRY)
+            && isSupportedBaseType(baseType, SUPPORTED_TYPES) ?
+            createNode(field.getDisplayContextParameter(), node.asText(), baseType, fieldPath) :
+            node;
+    }
+
+    @Override
+    protected JsonNode executeCollection(JsonNode collectionNode, CommonField caseViewField, String fieldPath) {
+        final BaseType collectionFieldType = BaseType.get(caseViewField.getFieldType().getCollectionFieldType().getType());
+
+        if (hasDisplayContextParameterType(caseViewField.getDisplayContextParameter(), DisplayContextParameterType.DATETIMEENTRY)
+            && isSupportedBaseType(collectionFieldType, SUPPORTED_TYPES)) {
+            ArrayNode newNode = MAPPER.createArrayNode();
+            collectionNode.forEach(item -> {
+                JsonNode newItem = item.deepCopy();
+                ((ObjectNode)newItem).replace(CollectionValidator.VALUE,
+                    createNode(caseViewField.getDisplayContextParameter(), item.get(CollectionValidator.VALUE).asText(), collectionFieldType, fieldPath));
+                newNode.add(newItem);
+            });
+
+            return newNode;
+        }
+
+        return collectionNode;
+    }
+
+    private TextNode createNode(String displayContextParameter, String valueToConvert, BaseType baseType, String fieldPath) {
+        String format = getDisplayContextParameterOfType(displayContextParameter, DisplayContextParameterType.DATETIMEENTRY)
+            .map(DisplayContextParameter::getValue)
+            .orElseThrow(() -> new DataProcessingException().withDetails(
+                String.format("Unable to obtain datetime format for field %s with display context parameter %s",
+                    fieldPath,
+                    displayContextParameter)
+            ));
+        try {
+            if (baseType == BaseType.get(DATETIME)) {
+                return new TextNode(dateTimeFormatParser.convertDateTimeToIso8601(format, valueToConvert));
+            } else {
+                return new TextNode(dateTimeFormatParser.convertDateToIso8601(format, valueToConvert));
+            }
+        } catch (Exception e) {
+            throw new DataProcessingException().withDetails(
+                String.format("Unable to process field %s with value %s. Expected format: %s",
```

Comment:
If I'm not wrong here we are failing the case update operation if the date field is not in the format specified in the DisplayContextParameter. This won't be an issue with the UI because it's going to send us the date fields in the correct format. 
But what about Services that are invoking the update event programmatically? They will still be passing unformatted date fields probably? will those calls throw exception? 
In case the date is already in ISO8601 format, does the `dateTimeFormatParser.convertDateTimeToIso8601` throws exception? 
We could think of allowing date fields that are already in ISO8601 format rather than throwing exception. 
We need to speak with BAs to understand the behaviour in this situation and state it clearly in the ticket for future reference